tests/std/tests/GH_001964_constexpr_generate_canonical/test.cpp

```diff
@@ -0,0 +1,57 @@
+#include <cassert>
+#include <cmath>
+#include <cstdint>
+#include <random>
+#include <utility>
+
+using namespace std;
+
+int naive_iterations(const int bits, const uint64_t gmin, const uint64_t gmax) {
+    // Naive implementation of [rand.util.canonical]. Note that for large values of range, it's possible that
+    // log2(range) == bits when range < 2^bits. This can lead to incorrect results, so we can't use this function as
+    // a reference for all values.
+
+    const double range = static_cast<double>(gmax) - static_cast<double>(gmin) + 1.0;
+    return max(1, static_cast<int>(ceil(static_cast<double>(bits) / log2(static_cast<double>(range)))));

```

Comment:
`static_cast<double>(range)` is unnecessary as we already have `const double range`.